sklearn/mixture/_gaussian_mixture_ic.py

```diff
@@ -503,7 +519,15 @@ def fit(self, X, y=None):
         )
 
         param_grid = list(ParameterGrid(param_grid))
-        param_grid_ag, param_grid = _process_paramgrid(param_grid)
+        param_grid_ag, param_grid = _process_paramgrid(
+            param_grid, self.n_init, self.label_init
+        )
+
+        if isinstance(random_state, int):
+            np.random.seed(random_state)
+            seeds = np.random.randint(1e8, size=len(param_grid))
```

Comment:
this should be something like `np.iinfo(np.int32).max` see [here](https://joblib.readthedocs.io/en/latest/auto_examples/parallel_random_state.html#fixing-the-random-state-to-obtain-deterministic-results)